src/api/RolesApi.ts

```diff
@@ -50,135 +43,170 @@ interface SPRole {
 }
 
 //* Format for request for adding a Role to a user */
-export interface ISubmitRole {
+interface ISubmitRole {
   /** The User to add the Role to */
   User: IPerson;
   /** The Role to add to the User */
   Role: RoleType;
 }
 
+//* Format for sending request to SP for adding a Role to a user */
 interface ISPSubmitRole {
   Id?: number;
   UserId: number;
   Title: RoleType;
 }
 
+/** Type for Map of User Roles grouped with key of Role */
+type IRolesByType = Map<RoleType, SPRole[]>;
+
+/** Type for Map of User Roles grouped with key of UserId */
+type IRolesByUser = Map<number, SPRole[]>;
+
 /** Test data for use in DEV environment -- mimics structure of Roles list in SharePoint */
-let testRoles: SPRole[] = [
-  {
-    Id: 1,
-    User: {
+let testRoles: SPRole[];
+
+/** The maxId of records in testRoles -- used for appending new roles in DEV env to mimic SharePoint */
+let maxId: number;
+
+// Only populate the testRoles and maxId if we are in DEV
+if (process.env.NODE_ENV === "development") {
+  testRoles = [
+    {
       Id: 1,
-      Title: "FORREST, GREGORY M CTR USAF AFMC AFLCMC/OZIC",
-      EMail: "me@example.com",
+      User: {
+        Id: 1,
+        Title: "FORREST, GREGORY M CTR USAF AFMC AFLCMC/OZIC",
+        EMail: "me@example.com",
+      },
+      Title: RoleType.ADMIN,
     },
-    Title: RoleType.ADMIN,
-  },
-  {
-    Id: 2,
-    User: {
+    {
       Id: 2,
-      Title: "PORTERFIELD, ROBERT D GS-13 USAF AFMC AFLCMC/OZIC",
-      EMail: "me@example.com",
+      User: {
+        Id: 2,
+        Title: "PORTERFIELD, ROBERT D GS-13 USAF AFMC AFLCMC/OZIC",
+        EMail: "me@example.com",
+      },
+      Title: RoleType.IT,
     },
-    Title: RoleType.IT,
-  },
-  {
-    Id: 3,
-    User: {
-      Id: 1,
-      Title: "FORREST, GREGORY M CTR USAF AFMC AFLCMC/OZIC",
-      EMail: "me@example.com",
+    {
+      Id: 3,
+      User: {
+        Id: 1,
+        Title: "FORREST, GREGORY M CTR USAF AFMC AFLCMC/OZIC",
+        EMail: "me@example.com",
+      },
+      Title: RoleType.IT,
     },
-    Title: RoleType.IT,
-  },
-];
+  ];
 
-/** The maxId of records in testRoles -- used for appending new roles in DEV env to mimic SharePoint */
-let maxId: number = testRoles.length;
+  maxId = testRoles.length;
+}
 
 /** Function used by calls in the DEV env to mimic a delay in processing
  *
  * @param milliseconds - Optional value to sleep -- Defaults to 500 if not specified
+ * @param resolveValue - Optional value to pass back as the reolved promise
  * @returns A Promise after the delayed amount of time
  */
-const sleep = (milliseconds?: number) => {
+const sleep = <T extends any>(
+  milliseconds?: number,
+  resolveValue?: T
+): Promise<T> => {
```

Comment:
Nice.  Updated the function to initialize the milliseconds and leverage it directly instead of adding a separate variable.  Also removed extending the type.  Kind of torn on moving it to a Utility library -- as might be OBE if we implement a "mock server".